File: solution/53/solution.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==0){
            return 0;
        }
        vector<int> datas(nums.size(), 0);
        int result=nums[0];
        datas[0] = nums[0];
        for(int i=1;i<nums.size();i++){
            datas[i] = max(nums[i], datas[i-1] + nums[i]);
            if(datas[i]>result){
                result = datas[i];
            }
        }
        return result;
    }
};

File: solution/53/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        datas = [nums[0]]
        result = nums[0]
        for i in xrange(1, len(nums)):
            datas[i] = max(nums[i], datas[i-1]+nums[i])
            if result < datas[i]:
                result = datas[i]
        return result